src/main/java/ChatServer.java

```diff
@@ -0,0 +1,75 @@
+import java.io.*;
+import java.net.ServerSocket;
+import java.net.Socket;
+
+public class ChatServer {
+    private ServerSocket serverSocket;
+    private UserIO io;
+
+    public ChatServer(ServerSocket serverSocket, UserIO io) {
+        this.serverSocket = serverSocket;
+        this.io = io;
+    }
+
+    public Socket makeConnection(ServerSocket serverSocket) {
+        try {
+            Socket socket = serverSocket.accept();
+            return socket;
+        } catch (IOException e) {
+            io.showOutput("Error in connecting socket");
+            e.printStackTrace();
```

Comment:
You should never need to return null. What you can do here is throw new UncheckedIOException(e) on line 19, and just let it propagate up the stack. Logging an error and continuing is dangerous. Instead you should fail fast